src/auth/auth.service.spec.ts

```diff
@@ -1,18 +1,123 @@
+import { JwtService } from '@nestjs/jwt';
 import { Test, TestingModule } from '@nestjs/testing';
+import { UserService } from 'src/user/user.service';
 import { AuthService } from './auth.service';
+import { faker } from '@faker-js/faker';
+import * as bcrypt from 'bcrypt';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { Role } from '@prisma/client';
 
 describe('AuthService', () => {
-  let service: AuthService;
+  let authService: AuthService;
+  let userService: UserService;
+
+  class MockRequest {
+    email: string;
+    password: string;
+  }
+
+  const mockJwtService = {
+    signAsync: jest.fn().mockImplementation((_request) => {
+      return Promise.resolve(faker.string.sample(64));
+    }),
+  };
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [AuthService],
+      providers: [
+        AuthService,
+        PrismaService,
+        UserService,
+        {
+          provide: JwtService,
+          useValue: mockJwtService,
+        },
+      ],
     }).compile();
 
-    service = module.get<AuthService>(AuthService);
+    jest.clearAllMocks();
+    authService = module.get<AuthService>(AuthService);
+    userService = module.get<UserService>(UserService);
   });
 
   it('should be defined', () => {
-    expect(service).toBeDefined();
+    expect(authService).toBeDefined();
+  });
+
+  describe('SignIn', () => {
+    it('should return access token', async () => {
+      //given
+      const request: MockRequest = {
+        email: faker.internet.email(),
+        password: faker.internet.password(),
+      };
+
+      //when
+      const accessToken = authService.signIn(request);
+
+      //then
+      expect(accessToken).toBeDefined();
+    });
+  });
+
+  describe('SignUp', () => {
+    it('should sign up user', async () => {
+      //given
+      const request: MockRequest = {
+        email: faker.internet.email(),
+        password: faker.internet.password(),
+      };
+
+      jest.spyOn(userService, 'createUser').mockImplementation((request) => {
+        return Promise.resolve({
+          id: faker.number.int(),
+          email: request.email,
+          password: request.password,
```

Comment:
didn't we fix it already? We don't want to include user password in the response

Reply:
Shouldn't service return password here, so controller can map it and 'ignore' it?

Reply:
If controller does not need to return a `password` field in a response, then there is no need to provide it to the controller. If `password` is not provided, controller won't have to ignore it, because there is nothing to ignore.

Service could return a `password` field if there was some other logic using this service method requiring `password` field for something, but I don't believe we have such logic here.